### DISCUSSION TITLE
Bring back the car model

### DISCUSSION THREAD

**Discussion by jhurliman** [foxglove/community#224] [Q&A]
Requiring external hosting of mesh markers is burdensome. It would be nice to bring back the embedded autonomous vehicle glTF, and allow referencing it with a special URL such as `foxglove://car`.

**Comment by amacneil** [comment]
Maybe we offer a few options in a dropdown or something?

Although, do people actually want built in models that are not their robot? I can't think of who would want this (besides us developing FG).

Sent via Superhuman ( ***@***.*** )

**Comment by jhurliman** [comment]
@amacneil I was thinking this would be for mesh markers, although it could be used for `PoseStamped` as well with a dropdown in the UI.

> do people actually want built-in models that are not their robot

We already supply a built-in model that is not their robot. It's a low poly cylinder / arrow thing. This request is to expand support beyond our single model to at least one more.

**Comment by amacneil** [comment]
Still don't understand who this would be useful for. Pretty big difference between supporting generic shapes and a "Chevvy Bolt" model.

**Comment by jhurliman** [comment]
@amacneil deeproute.ai

![image](https://user-images.githubusercontent.com/195374/149248766-494bd139-4463-4801-b57d-167dd0dca16f.png)

You can see in their promo video they use a generic car model that does not appear to be an accurate model of a fully rigged self-driving car. They thought a generic car model made for better data visualization than a purple cylinder driving down the road. To get to a car model in Studio, you can:

1. Acquire a 3D model of your vehicle, often derived from a CAD model or another very high poly source
2. Hire an artist to create a low poly rendition of this model
3. Convert the model to glTF format, ensuring you set the correct up axis and scale, the file size is reasonable, and no unsupported glTF extensions are enabled
4. Upload this model to a publicly accessible web server and configure the correct CORS headers. Alternatively, if you want to keep your model private you need to run a private web server hosting this file
5. Point your mesh_resource at this public or private URL

With an embedded car model, you could get to the same visual fidelity of the deeproute.ai promo video by:

1. Point your mesh_resource at `foxglove://car`

**Comment by rgov** [comment]
What about making it easy to select Ignition Fuel models? If you can support loading rendering SDF / .obj files that is.

https://app.ignitionrobotics.org/OpenRobotics/fuel/models/Hatchback%20red

**Comment by defunctzombie** [comment]
If we have some notion of an asset manager within the app we could allow adding datasets like the above referenced model dataset.
